src/ChannelServer/World/Entities/Components/Buffs.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Copyright (c) Aura development team - Licensed under GNU GPL
+// For more information, see licence.txt in the main folder
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Melia.Channel.Network;
+using Melia.Channel.World.Entities;
+using Melia.Shared.Const;
+using Melia.Shared.EntityComponents;
+
+namespace Melia.Channel.World
+{
+	/// <summary>
+	/// Character buffs.
+	/// </summary>
+	public class Buffs : IUpdatableComponent
+	{
+		private readonly Dictionary<BuffId, Buff> _buffs = new Dictionary<BuffId, Buff>();
+
+		/// <summary>
+		/// The owner of this object.
+		/// </summary>
+		public ICombatEntity Owner { get; }
+
+		/// <summary>
+		/// Creates new instance for character.
+		/// </summary>
+		/// <param name="entity"></param>
+		public Buffs(ICombatEntity entity)
+		{
+			this.Owner = entity;
+		}
+
+		/// <summary>
+		/// Returns the amount of buffs in the collection.
+		/// </summary>
+		public int Count { get { lock (_buffs) return _buffs.Count; } }
+
+		/// <summary>
+		/// Adds given without updating the client. Replaces existing
+		/// buffs.
+		/// </summary>
+		/// <param name="buff"></param>
+		public void AddSilent(Buff buff)
+		{
+			lock (_buffs)
+				_buffs[buff.Id] = buff;
+		}
+
+		/// <summary>
+		/// Adds given buff and updates the client.
+		/// </summary>
+		/// <param name="buff"></param>
+		public void Add(Buff buff)
+		{
+			buff.IncreaseOverbuff();
+			this.AddSilent(buff);
+			var handler = ChannelServer.Instance.BuffHandlers.GetBuff(buff.Id);
+			handler.OnStart(buff);
+			Send.ZC_BUFF_ADD(this.Owner, buff);
+		}
+
+		/// <summary>
+		/// Removes buff with given id, returns false if it
+		/// didn't exist. Doesn't update the client.
+		/// </summary>
+		/// <param name="buffId"></param>
+		/// <returns></returns>
+		public bool RemoveSilent(BuffId buffId)
+		{
+			lock (_buffs)
+				return _buffs.Remove(buffId);
+		}
+
+		/// <summary>
+		/// Removes buff with given id, returns false if it
+		/// didn't exist. Updates the client on success.
+		/// </summary>
+		/// <param name="buffId"></param>
+		/// <returns></returns>
+		public bool Remove(BuffId buffId)
+		{
+			var buff = this.Get(buffId);
+			if (buff != null)
+			{
+				this.Remove(buff);
```

Comment:
You should return true here, as that's the expected return value for when the buff _was_ removed.

Reply:
```
		 public bool Remove(BuffId buffId)
		{
			var buff = this.Get(buffId);
			if (buff != null)
				return this.Remove(buff);
			return false;
		}
```

Reply:
Yea, that will work.